File: dashboard-back-end/api_soil_breath/Entity/Cultura.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace api_soil_breath.Entity
{
    [Table("Cultura")]
    public class Cultura
    {
        [Column("id")]
        public int Id { get; set; }

        [Column("nome")]
        public string Nome { get; set; }

        [Column("nitrogenio")]
        public double Nitrogenio { get; set; }

        [Column("fosforo")]
        public double Fosforo { get; set; }

        [Column("potassio")]
        public double Potassio { get; set; }

        [Column("umidade")]
        public double Umidade { get; set; }
    }
}

File: dashboard-front-end/src/api/services/cultura.ts
import { apiClient, ApiResponse } from '../client';
import { API_ENDPOINTS } from '../config';

// Interface baseada no DTO C# CulturaDTO
export interface Cultura {
  id: number;
  nome: string;
  nitrogenio: number;
  fosforo: number;
  potassio: number;
  umidade: number;
  temperatura: number;
}

// Interface baseada no DTO C# AptidaoDTO
export interface AptidaoCultura {
  culturaId: number;
  culturaNome: string;
  nitrogenioPct: number;
  fosforoPct: number;
  potassioPct: number;
  mediaPct: number;
}

// Interface baseada no DTO C# AptidaoSoloDTO
export interface AptidaoSolo {
  soloId: number;
  soloIdentificacao: string;
  nitrogenio: number;
  fosforo: number;
  potassio: number;
  aptidoes: AptidaoCultura[];
}

// Serviço de Cultura
export const culturaService = {
  // Buscar todas as culturas
  async getCulturas(): Promise<ApiResponse<Cultura[]>> {
    return apiClient.get<Cultura[]>(API_ENDPOINTS.CULTURAS);
  },

  // Buscar uma cultura específica
  async getCulturaById(id: number): Promise<ApiResponse<Cultura>> {
    return apiClient.get<Cultura>(API_ENDPOINTS.CULTURA_BY_ID(id));
  },

  // Calcular aptidões de um solo para todas as culturas
  async getAptidoesPorSolo(soloId: number): Promise<ApiResponse<AptidaoSolo>> {
    return apiClient.get<AptidaoSolo>(`${API_ENDPOINTS.CULTURAS}/aptidao/solo/${soloId}`);
  },
};
